pysmt/oracles.py

```diff
@@ -327,6 +329,75 @@ def walk_function(self, formula, args):
 
 # EOC FreeVarsOracle
 
+class AtomsOracle(pysmt.walkers.DagWalker):
+    """This class returns the set of theory atoms involved in a formula"""
+    def __init__(self, env=None):
+        pysmt.walkers.DagWalker.__init__(self, env=env)
+
+        # Clear the mapping function
+        self.functions.clear()
+
+        # We have teh following categories for this walker.
+        #
+        # - Boolean operators, e.g. and, or, not...
+        # - Theory operators, e.g. +, -, bvshift
+        # - Theory relations, e.g. ==, <=
+        # - ITE terms
+        # - Symbols
+        # - Constants
+        #
+        for elem in op.ALL_TYPES:
+            if elem in op.BOOL_CONNECTIVES:
+                self.functions[elem] = self.walk_bool_op
+            elif elem in op.CONSTANTS:
+                self.functions[elem] = self.walk_constant
+            elif elem in op.RELATIONS:
+                self.functions[elem] = self.walk_theory_relation
+            elif elem in op.BV_OPERATORS or elem in op.LIRA_OPERATORS:
+                self.functions[elem] = self.walk_theory_op
+            else:
+                self.functions[elem] = self.walk_error
+
+        self.functions[op.SYMBOL] = self.walk_symbol
+        self.functions[op.FUNCTION] = self.walk_theory_op
+        self.functions[op.ITE] = self.walk_ite
+
+        # Check that no operator in undefined
+        assert self.is_complete(verbose=True)
+
+
+    def get_atoms(self, formula):
+        """Returns the set of atoms appearing in the formula."""
+        return self.walk(formula)
+
+    def walk_bool_op(self, formula, args):
+        return frozenset(x for a in args for x in a)
+
+    def walk_theory_relation(self, formula, args):
+        return frozenset([formula])
+
+    def walk_theory_op(self, formula, args):
+        return None
+
+    def walk_symbol(self, formula, args):
+        if formula.is_symbol(types.BOOL):
```

Comment:
This is returning also boolean symbols, but the description said that it returns theory atoms.